### PR TITLE
feat: add downvote events

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- add downvote events (mirroring upvote events)
- added to feed, modal, page and companion
- refactor: removed upvote logic from `postEventName` in favor of `AnalyticsEvent` usage, this was due to complicated logic of resolving event name when `downvoted` and `upvoted` properties were both passed through `update` object of `postEventName`

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| New | downvote post | extra: { origin } |
| New | remove post downvote | extra: { origin } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1466 #done

### CODE REVIEW

File: packages/extension/src/companion/CompanionMenu.tsx [hunk 1]
```diff
@@ -89,14 +89,36 @@ export default function CompanionMenu({
     removeCompanionHelper,
     toggleCompanionExpanded,
   } = useCompanionActions({
-    onBookmarkMutate: () => updatePost({ bookmarked: true }),
-    onRemoveBookmarkMutate: () => updatePost({ bookmarked: false }),
-    onUpvotePostMutate: () => updatePost(mutationHandlers.upvote(post)),
+    onBookmarkMutate: () =>
+      updatePost({
+        update: { bookmarked: true },
+        event: postEventName({ bookmarked: true }),
```
Comment (rebelchris) [inline packages/extension/src/companion/CompanionMenu.tsx:95]: Maybe we should convert these to `AnalyticsEvent` as well? Just so it's uniform and just remove `postEventName?

If too much work never mind though.

File: packages/extension/src/companion/CompanionMenu.tsx [hunk 2]
```diff
@@ -60,14 +60,14 @@ export default function CompanionMenu({
...
+  const updatePost = async ({ update, event }) => {
     const oldPost = post;
     setPost({
       ...post,
       ...update,
     });
     trackEvent(
-      postAnalyticsEvent(postEventName(update), post, {
+      postAnalyticsEvent(event, post, {
         extra: { origin: 'companion context menu' },
```
Comment (rebelchris) [inline packages/extension/src/companion/CompanionMenu.tsx:71]: Best to use `Origin.{origin}` here from the enum export

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 3]
```diff
@@ -149,6 +149,7 @@ export default function PostOptionsMenu({
           null,
         );
       },
+      origin,
```
Comment (rebelchris) [inline packages/shared/src/components/PostOptionsMenu.tsx:152]: Just to check what's the origin referring to here?

File: packages/shared/src/components/post/PostActions.tsx [hunk 4]
```diff
@@ -99,8 +98,20 @@ export function PostActions({
     }
 
     if (post.downvoted) {
+      trackEvent(
+        postAnalyticsEvent(AnalyticsEvent.RemovePostDownvote, post, {
+          extra: { origin },
```
Comment (rebelchris) [inline packages/shared/src/components/post/PostActions.tsx:103]: Just to verify again, this is origin that comes from a component right?

**Review by rebelchris — APPROVED** [review]
Awesome. looks complete.
Please see my additional note on cleanup (which I agree with you did)

File: packages/shared/src/components/post/PostActions.tsx (hunk 4 above)
Comment (capJavert) [reply packages/shared/src/components/post/PostActions.tsx:103]: Same as https://github.com/dailydotdev/apps/pull/1899#discussion_r1242166049

File: packages/extension/src/companion/CompanionMenu.tsx (hunk 2 above)
Comment (capJavert) [reply packages/extension/src/companion/CompanionMenu.tsx]: Added in c3f306058d206e9b4c8efb75c6f2a4ec1c208c46

File: packages/extension/src/companion/CompanionMenu.tsx (hunk 1 above)
Comment (capJavert) [reply packages/extension/src/companion/CompanionMenu.tsx]: Added in 79cf70a42d4d92aa26471f367d414b17aa0b6afb

**Review by rebelchris — APPROVED** [review]
Nice, thanks for the retro fixes

**Pull request merged**